shared_memory_client.py

```diff
@@ -0,0 +1,170 @@
+from pickle import loads, dumps
+from tcp_communication import TCP_Connection
+from messages import Request, Response
+import SharedMemoryWrapper
+
+
+class SharedMemoryClient:
```

Comment:
You should create an interface for interacting with shared memory (will contain getByCounter, getLatest, publish ,etc...).
This class should implement this interface, as well as the class that interacts with shared memory on a local machine.

Reply:
I'm not sure I understood you correctly...
All the functions you described, like getLatest or getByCounter, are already implemented in the SharedMemoryWrapper module, which is written in cpp. All you need to do is just calling the function you want to use, and give it the correct arguments. For example, if you want to publish to a remote machine over UDP, you just need to put the connection info in the config file, and call the function getConnection, and call the publish function for example on the connection object.